pio-parser/src/lib.rs

```diff
@@ -189,7 +189,13 @@ impl<'i> ParsedOperands<'i> {
             },
             ParsedOperands::SET { destination, data } => InstructionOperands::SET {
                 destination: *destination,
-                data: data.reify(state) as u8,
+                data: {
+                    let arg = data.reify(state);
+                    if arg < 0 || arg > 0x1f {
+                        panic!("SET argument out of range: {}", arg);
+                    }
+                    data.reify(state) as u8
```

Comment:
seems strange to call `data.reify(state)` when you've got the binding `arg` that you've used for the last few lines.

Reply:
Of course! I didn't mean to repeat the call to reify, this was a plain oversight.